packages/realm-react/README.md

```diff
@@ -323,8 +323,149 @@ const SomeComponent = () => {
 }
 ```
 
+### Authentication Hooks
 
-#### Multiple Realms
+The following hooks can be used to authenticate users in your application.  They return authentication operations and a single result object which can be read to track the progress of the current result. More information about the specific auth methods can be found in the [Authenticate Users](https://www.mongodb.com/docs/realm/sdk/react-native/manage-users/authenticate-users) Documentation.
+
+## `useAuth`
+These hooks would typically be used in the `fallback` component of the `UserProvider`
+
+### `result`
+The result has the following structure:
+```tsx
+{
+  /**
+   * The current state of the operation.
+   * Enumerated by OperationState
+   */
+  state, // "not-started", "pending", "success", "error"
+
+  /**
+   * Convenience accessors, so users can write e.g. `loginResult.pending`
+   * instead of `loginResult.state === OperationState.Pending`
+   */
+  pending, // true or false
+  success, // true or false
+
+  /**
+   * The error returned from the operation, if any. This will only be populated
+   * if `state === OperationState.Error`, and will be cleared each time the
+   * operation is called.
+   */
+  error // Error based object or undefined
+}
+```
+
+This can be used to manage the state of the current login operation.
+
+### `logIn`
+Log in with a Realm.Credentials instance. This allows login with any authentication mechanism supported by Realm. If this is called when a user is currently logged in, it will switch the user.  Typically the other methods from `useAuth` would be used.
+```tsx
+const {logIn, result} = useAuth();
+const performLogin = () => {
+  logIn(Realm.Credential.anonymous())
+}
+```
+
+### `logInWithAnonymous`
+Log in with the Anonymous authentication provider.
+```tsx
+const {logInWithAnonymous, result} = useAuth();
+const performLogin = () => {
+  logInWithAnonymous();
+};
+```
+
+### `logInWithApiKey`
+Log in with an API key.
+```tsx
+const {logInWithApiKey, result} = useAuth();
+const performLogin = () => {
+  const key = getApiKey(); // user defined function
+  logInWithApiKey(key);
+};
+```
+
+### `logInWithEmailPassword`
+Log in with Email/Password.
+```tsx
+const {logInWithEmailPassword, result} = useAuth();
+const [email, setEmail] = useState();
+const [password, setPassword] = useState();
+
+const performLogin = () => {
+  logInWithEmailPassword({email, password});
+};
+```
+
+### `logInWithJWT`
+Log in with a JSON Web Token (JWT).
+```tsx
+const {logInWithJWT, result} = useAuth();
+
+const performLogin = () => {
+  const token = authorizeWithCustomerProvider(); //user defined function
```

Comment:
Alternatively take this as an argument for the function. Just a thought.

```suggestion
const performLogin = (token) => { // a token provided by some other authentication service
```

Reply:
Guess it depends on what is calling this.  Honestly, I think most of these examples should be clarified in the mdb docs with a bit more context.